src/cmakeWorkspaceFolder.ts

```diff
@@ -214,30 +53,85 @@ export class CMakeWorkspaceFolderController implements vscode.Disposable {
      * - Where we search for variants
      * - Where we search for workspace-local kits
      */
-    private _activeFolder?: CMakeWorkspaceFolder;
-    get activeFolder() {
-        return this._activeFolder;
+
+    get activeFolderPath(): string | undefined {
+        return this.activeCMakeProject?.folderPath;
+    }
+    /**
+     * The name of the folder for this CMakeProject instance
+     */
+    get activeFolderName(): string | undefined {
+        return this.activeCMakeProject?.folderName;
     }
-    setActiveFolder(ws: vscode.WorkspaceFolder | undefined) {
-        if (ws) {
-            this._activeFolder = this.get(ws);
-        } else {
-            this._activeFolder = undefined;
+
+    private activeCMakeProject: CMakeProject | undefined;
+    setActiveCMakeProject(workspaceFolder?: vscode.WorkspaceFolder, openEditor?: vscode.TextEditor, folderName?: string): CMakeProject | undefined {
```

Comment:
It's not common for a 'set'-named method to return something.  I would call it 'updateActiveCMakeProject' and require any callers that need the updated value to call the 'getActiveCMakeProject' method afterwards.  (There is a hierarchy of functions called 'set' that should be renamed - everyone that calls this method, etc)

Also, you don't need to include "CMake" in the names of these functions.  `getActiveProject` is explanatory enough I think.